react-file-uploads/client/app.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      caption: ''
+    };
+    this.fileInputRef = React.createRef();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleCaptionChange = this.handleCaptionChange.bind(this);
+  }
+
+  handleCaptionChange(event) {
+    this.setState({ caption: event.target.value });
+  }
+
+  handleSubmit(event) {
+    /**
+     * Prevent the browser's default behavior for form submissions.
+     *
+     * Create a `new` FormData object.
+     *
+     * Append two entries to the form data object you created.
+     *   1. "caption" the value of this.state.caption
+     *   2. "image" the value of this.fileInputRef.current.files[0]
+     *
+     * Use fetch() to send a POST request to /api/uploads. The body
+     * should be the form data object you created (not a JSON string)
+     * Headers are not necessary as fetch will use the correct Content-Type
+     * automatically (multiplart/form-data).
+     *
+     * Then 😉,
+     *   parse the JSON response body
+     * Then 😉,
+     *   log the parsed response body
+     *   set the caption state back to an empty string
+     *   assign null to this.fileInputRef.current.value to clear the file
+     * Catch any error in the promise chain.
+     *
+     * References:
+     * https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData
+     * https://developer.mozilla.org/en-US/docs/Web/API/FormData/append
+     * https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_a_file
+     * https://reactjs.org/docs/uncontrolled-components.html#the-file-input-tag
+     * https://reactjs.org/docs/refs-and-the-dom.html
+     */
+  }
```

Comment:
the client code also needs to be completed

Reply:
<img width="925" alt="Screen Shot 2022-09-08 at 5 04 15 PM" src="https://user-images.githubusercontent.com/106217355/189246110-af4e7f65-a020-445d-82a3-4482c063a2ac.png">
![Screen Shot 2022-09-08 at 5 04 11 PM](https://user-images.githubusercontent.com/106217355/189246115-f140ad75-6627-4f5f-a0b6-76a81b4e58f4.png)
![Screen Shot 2022-09-08 at 5 04 07 PM](https://user-images.githubusercontent.com/106217355/189246119-fd305cc0-bcb4-44ad-a25c-d0507b969b7a.png)